modules/ml/src/main/java/org/apache/ignite/ml/math/primitives/matrix/AbstractMatrix.java

```diff
@@ -918,4 +918,38 @@ else if (mean > threshold + iv)
     @Override public String toString() {
         return "Matrix [rows=" + rowSize() + ", cols=" + columnSize() + "]";
     }
+
+    /** {@inheritDoc} */
+    @Override public double determinant() {
+        //TODO: IGNITE-11192, use nd4j
+        LUDecomposition dec = new LUDecomposition(this);
+        double res = dec.determinant();
+        dec.destroy();
```

Comment:
Does it make sense to wrap it into `try-finally`?

Reply:
Yeah, great idea. Replace destroyable iface to autocloseable.